pkg/k8s/state_into_spec_annotation.go

```diff
@@ -19,24 +19,19 @@ import (
 	"strings"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-func ValidateOrDefaultStateIntoSpecAnnotation(obj *unstructured.Unstructured) error {
+// ValidateOrDefaultStateIntoSpecAnnotation validates the value of the
+// 'state-into-spec' annotation if it is set and defaults the annotation to the
+// passed in defaultValue if it is unset.
+func ValidateOrDefaultStateIntoSpecAnnotation(obj metav1.Object, gvk schema.GroupVersionKind, defaultValue string) error {
```

Comment:
Nit: If we pass `obj client.Object`, we can avoid passing the GVK also I believe.  I'm not sure we even need the GVK anyway!

Reply:
Just for self-education purpose, are `metav1.Object`, `client.Object`, and `unstructured.Unstructured` just interfaces for the KRM object of different granularities?

Also with https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1178, a GVK is no longer needed. I'll need to change it after a rebase.

Reply:
Rebased.

Didn't change it to `client.Object` as further refactoring will be needed.